docs/en/reference/working-with-associations.rst

```diff
@@ -500,8 +500,27 @@ and the "remove" operation.
         //...
     }
 
-Even though automatic cascading is convenient it should be used
-with care. Do not blindly apply cascade=all to all associations as
+Since ``cascade: persist`` is configured for the ``User#commentsAuthored``
+association, you can now create a user and persist his comment as follows:
+
+.. code-block:: php
+
+    <?php
+    $user = new User();
+    $myFirstComment = new Comment();
+    $user->addComment($myFirstComment);
+    
+    $em->persist($user);
+    $em->flush();
+
+.. note::
+
+    This code does not associate the newly created comment with the user.
+    To achieve this, you *always* have to call ``$myFirstComment->setUser($user);`` –
+    no matter if ``cascade: persist`` is set or not.
```

Comment:
@ThomasLandauer Let me clarify:

Let's say that you have an association tree `A -> B -> C`, where `A` "owns" `B`, and `B` owns `C`. You don't really want to expose `B` and `C` to the outside world, so they need to stay hidden in the implementation.

The various parts of the app should therefore only touch `A`, so you'd have `$em->persist($a)` and `$em->remove($a)`, but never touch instances of `B`, `C`.

This is the perfect use-case scenario for cascade operations, since the ORM will therefore interact with child elements only via mappings, and not via explicit API usage in your application layer (which would cause references of `B` and `C` to leak around).

There are other use-cases, but this is the main one.

As for "why does this not just work out of the box?", the reason is that otherwise accidents may happen when instances leak into the graph, and they get either persisted or removed automagically. You want to retain control over this, so an explicit cascade operation has to happen here.

Also, cascades are expensive (performance-wise) and should be used sparingly.

Makes sense?